physics/module_bl_mynn.F90

```diff
@@ -3793,73 +4032,146 @@ END SUBROUTINE mynn_tendencies
     REAL, DIMENSION( kts:kte+1,nchem), INTENT(IN) :: s_awchem
-    REAL, DIMENSION( ndvel ), INTENT(INOUT) :: vd1
-
+    REAL, DIMENSION( ndvel ), INTENT(IN) :: vd1
+    REAL, INTENT(IN) :: emis_ant_no,frp_mean,pblh
+    LOGICAL, INTENT(IN) :: enh_vermix
 !local vars
 
-    REAL, DIMENSION(kts:kte) :: dtz,vt,vq
-    REAL, DIMENSION(1:kte-kts+1) :: a,b,c,d
+    REAL, DIMENSION(kts:kte) :: dtz
+    REAL, DIMENSION(1:kte-kts+1) :: a,b,c,d,x
     REAL :: rhs,gfluxm,gfluxp,dztop
-    REAL :: t,esl,qsl
+    REAL :: t,esl,qsl,dzk
+    REAL :: hght 
+    REAL :: khdz_old, khdz_back
     INTEGER :: k,kk
     INTEGER :: ic  ! Chemical array loop index
-    REAL, DIMENSION( kts:kte, nchem ) :: chem_new
+    
+    INTEGER, SAVE :: icall
+
+    REAL, DIMENSION(kts:kte) :: rhoinv
+    REAL, DIMENSION(kts:kte+1) :: rhoz,khdz
+    REAL, PARAMETER :: no_threshold    = 0.1
+    REAL, PARAMETER :: frp_threshold   = 0.0
+    REAL, PARAMETER :: pblh_threshold  = 250.0
 
     dztop=.5*(dz(kte)+dz(kte-1))
 
     DO k=kts,kte
        dtz(k)=delt/dz(k)
     ENDDO
 
+    !Prepare "constants" for diffusion equation.
+    !khdz = rho*Kh/dz = rho*dfh
+    rhoz(kts)  =rho(kts)
+    rhoinv(kts)=1./rho(kts)
+    khdz(kts)  =rhoz(kts)*dfh(kts)
+! JLS
+    khdz_old  = khdz(kts)
+    khdz_back = pblh * 0.15 / dz(kts)
+    IF ( enh_vermix ) THEN
+    IF ( pblh < pblh_threshold ) THEN
+       IF ( emis_ant_no > no_threshold ) THEN
+          khdz(k) = MAX(khdz(k),khdz_back)
+       ENDIF
+       IF ( frp_mean > frp_threshold ) THEN
+          khdz(k) = MAX(khdz(k),khdz_back)
+       ENDIF
+    ENDIF
+    ENDIF
+    DO k=kts+1,kte
+       rhoz(k)  =(rho(k)*dz(k-1) + rho(k-1)*dz(k))/(dz(k-1)+dz(k))
+       rhoz(k)  =  MAX(rhoz(k),1E-4)
+       rhoinv(k)=1./MAX(rho(k),1E-4)
+       dzk      = 0.5  *( dz(k)+dz(k-1) )
+       khdz(k)  = rhoz(k)*dfh(k)
+    ENDDO
+    khdz(kte+1)=rhoz(kte+1)*dfh(kte)
+
+    !stability criteria for mf
+    DO k=kts+1,kte-1
+       khdz(k) = MAX(khdz(k),  0.5*rho(k)* s_aw(k))
+       khdz(k) = MAX(khdz(k), -0.5*rho(k)*(s_aw(k)-s_aw(k+1)))
+
+          khdz_old  = khdz(k)
+          khdz_back = pblh * 0.15 / dz(k)
+       IF ( enh_vermix ) THEN
+       !Modify based on anthropogenic emissions of NO and FRP
+       IF ( pblh < pblh_threshold ) THEN
+          IF ( emis_ant_no > no_threshold ) THEN
+             khdz(k) = MAX(khdz(k),khdz_back)
+          ENDIF
+          IF ( frp_mean > frp_threshold ) THEN
+             khdz(k) = MAX(khdz(k),khdz_back)
+          ENDIF
+       ENDIF
+       ENDIF
+    ENDDO
+
   !============================================
   ! Patterned after mixing of water vapor in mynn_tendencies.
   !============================================
 
     DO ic = 1,nchem
        k=kts
 
-       a(1)=0.
-       b(1)=1.+dtz(k)*dfh(k+1)  - 0.5*dtz(k)*s_aw(k+1)
-       c(1)=-dtz(k)*dfh(k+1)    - 0.5*dtz(k)*s_aw(k+1)
-       d(1)=chem1(k,ic) + dtz(k) * -vd1(ic)*chem1(1,ic) - dtz(k)*s_awchem(k+1,ic)
+       !a(1)=0.
```

Comment:
Does this commented-out code need to be in the repository?

Reply:
Certainly not. I'll remove it.